addons/resource/models/utils.py

```diff
@@ -63,34 +63,48 @@ def filter_domain_leaf(domain, leaf_check):
 
     params:
         - domain: the domain that needs to be filtered
-        - leaf_check: the function that the field used in the leaf needs to verify to keep the leaf
+        - field_check: the function that the field name used in the leaf needs to verify to keep the leaf
+        - field_name_mapping: dictionary of the form {'field_name': 'new_field_name', ...}. Occurences of 'field_name'
+          in the first element of domain leaves will be replaced by 'new_field_name'.
     returns: The filtered version of the domain
     """
-    def _filter_domain_leaf_recursive(domain, leaf_check, operator=False):
-        """
-        return domain, rest_domain -> rest_domain should be empty if the operation is finished
-        """
-        if len(domain) == 0:
-            return ([], [])
-        if not operator:
-            first_elem = domain[0]
-            if first_elem not in DOMAIN_OPERATORS: #End of a current leaf
-                return ([], domain[1:]) if not leaf_check(first_elem[0]) else ([first_elem], domain[1:])
-            operator = first_elem
-            domain = domain[1:]
-
-        leaf_1, rest_domain = _filter_domain_leaf_recursive(domain, leaf_check)
-        if operator == NOT_OPERATOR:
-            return ([operator, *leaf_1], rest_domain) if leaf_1 else ([], rest_domain)
-        leaf_2, rest_domain = _filter_domain_leaf_recursive(rest_domain, leaf_check)
-        if leaf_1 == [] or leaf_2 == []:
-            return ((leaf_1 or leaf_2), rest_domain)
-        return  ([operator, *leaf_1, *leaf_2], rest_domain)
-
+    stack = [] # stack of elements (leaf or operator) to conserve (reverse it gives a domain)
+    ignored_elem = [] # history of ignored element in the domain (not added to the stack)
```

Comment:
```suggestion
    ignored_elems = [] # history of ignored elements in the domain (not added to the stack)
```
Or was it meant with no s? `was_ignored_stack` maybe (and `res_elems_stack` then)?